[WIP] Fix PHP notices with standard PSR-2.

ChristopheBoucaut opened the pull request:
I get this notice with PHP 5.6.11 and standard PSR-2.

With this code : 

```php
<?php

namespace Foo;

/**
 * TTTTT
 */
class A
```

I have :
```PHP Notice:  Undefined offset: 22 in /usr/share/pear/PHP/CodeSniffer/Standards/PSR2/Sniffs/Classes/ClassDeclarationSniff.php on line 145```
```PHP Notice:  Undefined offset: 12 in /usr/share/pear/PHP/CodeSniffer/Tokenizers/PHP.php on line 601```

aik099 commented:
You get what notice? You haven't mentioned actual notice message you're getting.

aik099 commented:
Also a test needs to be added to show that the issue has been fixed.

ChristopheBoucaut commented:
With this code : 

```php
<?php

namespace Foo;

/**
 * TTTTT
 */
class A
```

I have :
```PHP Notice:  Undefined offset: 22 in /usr/share/pear/PHP/CodeSniffer/Standards/PSR2/Sniffs/Classes/ClassDeclarationSniff.php on line 145```

I just realize, if I check without my fix, I get also :
```PHP Notice:  Undefined offset: 12 in /usr/share/pear/PHP/CodeSniffer/Tokenizers/PHP.php on line 601```

I will try for write a test today.

ChristopheBoucaut commented:
Current problem with TU :

```
There was 1 failure:

1) PSR2_Tests_Classes_ClassDeclarationUnitTest::testSniff
Fixed version of ClassDeclarationUnitTest.inc does not match expected version in ClassDeclarationUnitTest.inc.fixed; the diff is
--- CodeSniffer/Standards/PSR2/Tests/Classes/ClassDeclarationUnitTest.inc.fixed
+++ PHP_CodeSniffer
@@ -138,5 +138,3 @@
 } // C1
 
 class Test
-{
-}
```

Problem with the test on the fixer.
I try to fix it this week-end.

gsherwood commented:
The fixer shouldn't be fixing syntax errors for you. All sniffs and fixes should ignore the incomplete class statement, as they do for other incomplete control structures.

Some of the sniffs do produce syntax error warnings, while others do not. There is actually a generic syntax checking sniff that runs `php -l` over the code to report any syntax errors, so that's the preferred way of getting syntax checking into your standard these days.

gsherwood commented:
I've pushed a change so the PSR2 standard checks for the syntax error before processing. The parent class already generates the syntax warnings, so there is no need for it to also generate them.

So now you get:
```
FILE: temp.php
-------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------
 4 | WARNING | Possible parse error: class missing opening or closing brace (PSR2.Classes.ClassDeclaration.MissingBrace)
 4 | WARNING | Possible parse error: class missing opening or closing brace (Squiz.Classes.ValidClassName.MissingBrace)
-------------------------------------------------------------------------------------------------------------------------
```

ChristopheBoucaut commented:
Oh ! thank you ! I prefer your solution.
